admin/spam/views.py

```diff
@@ -1,47 +1,106 @@
-    except EmptyPage:
-        page = paginator.page(paginator.num_pages)
-    context = {
-        'spam': map(serialize_comment, page),
-        'page': page,
-        'page_number': page_number,
-    }
-    return render(request, 'spam/spam.html', context)
+    def get_queryset(self):
+        self.status = self.request.GET.get('status', u'1')
+        query = (
+            Q('reports', 'ne', {}) &
+            Q('reports', 'ne', None) &
+            Q('spam_status', 'eq', int(self.status))
+        )
+        return Comment.find(query).sort(self.ordering)
 
+    def get_context_data(self, **kwargs):
+        queryset = kwargs.pop('object_list', self.object_list)
+        page_size = self.get_paginate_by(queryset)
+        paginator, page, queryset, is_paginated = self.paginate_queryset(
+            queryset, page_size)
+        return {
+            'spam': map(serialize_comment, queryset),
+            'page': page,
+            'status': self.status,
+            'page_number': page.number
+        }
 
-@login_required
-def spam_detail(request, spam_id):
-    context = {
-        'comment': serialize_comment(Comment.load(spam_id)),
-        'page_number': request.GET.get('page', 1),
-    }
-    return render(request, 'spam/comment.html', context)
+
+class SpamDetail(FormView):
+    form_class = ConfirmForm
+    template_name = 'spam/comment.html'
+    spam_id = None
+    page = 1
+
+    def __init__(self):
+        self.item = None
+        super(SpamDetail, self).__init__()
+
+    @method_decorator(login_required)
+    def get(self, request, *args, **kwargs):
+        try:
+            context = self.get_context_data(**kwargs)
+        except AttributeError:
+            return page_not_found(request)  # TODO: 1.9 update to have exception with node/user 404.html will be added
+        self.page = request.GET.get('page', 1)
+        context['page_number'] = self.page
+        context['form'] = self.get_form()
+        return self.render_to_response(context)
+
+    @method_decorator(login_required)
+    def post(self, request, *args, **kwargs):
+        try:
+            context = self.get_context_data(**kwargs)
+        except AttributeError:
+            return page_not_found(request)  # TODO: 1.9 update to have exception
+        self.page = request.GET.get('page', 1)
+        context['page_number'] = self.page
+        context['form'] = self.get_form()
+        if context['form'].is_valid():
+            return self.form_valid(context['form'])
+        else:
+            return render(request, self.template_name, context=context)
+        # return super(SpamDetail, self).post(request, *args, **kwargs)
+
+    def get_context_data(self, **kwargs):
+        self.spam_id = kwargs['spam_id']
+        self.item = Comment.load(self.spam_id)
+        kwargs = super(SpamDetail, self).get_context_data(**kwargs)
+        try:
+            kwargs['comment'] = serialize_comment(self.item)
+        except AttributeError:
```

Comment:
You shouldn't need to catch this and then rethrow, should you? It should just bubble.